airflow/dags/unzip_and_validate_gtfs_schedule/unzip_gtfs_schedule.py

```diff
@@ -0,0 +1,191 @@
+# ---
+# python_callable: unzip
+# provide_context: true
+# ---
+import os
+import logging
+import pendulum
+import zipfile
+
+from typing import ClassVar, List, Optional
+from pydantic import Field
+
+from calitp.storage import (
+    fetch_all_in_partition,
+    GTFSFeedExtractInfo,
+    get_fs,
+    GTFSFeedType,
+    PartitionedGCSArtifact,
+    ProcessingOutcome,
+)
+
+# TODO: what level?
+logging.basicConfig(format="%(levelname)s:%(message)s", level=logging.INFO)
+
+# SCHEDULE_UNZIPPED_BUCKET = os.getenv("CALITP_BUCKET__GTFS_SCHEDULE_UNZIPPED")
+# SCHEDULE_RAW_BUCKET = os.getenv("CALITP_BUCKET__GTFS_SCHEDULE_RAW")
+# for testing:
+SCHEDULE_RAW_BUCKET = "test-calitp-gtfs-schedule-raw"
+SCHEDULE_UNZIPPED_BUCKET = "test-calitp-gtfs-schedule-unzipped"
+
+
+class GTFSScheduleFeedFile(PartitionedGCSArtifact):
+    bucket: ClassVar[str] = SCHEDULE_UNZIPPED_BUCKET
+    partition_names: ClassVar[List[str]] = ["dt", "base64_url", "ts"]
+    ts: pendulum.DateTime
+    base64_url: str
+    zipfile_path: str
+    original_filename: str
+
+    # if you try to set table directly, you get an error because it "shadows a BaseModel attribute"
+    # so set as a property instead
+    @property
+    def table(self) -> str:
+        # only replace slashes so that this is a mostly GCS-filepath-safe string
+        # if we encounter something else, we will address: https://cloud.google.com/storage/docs/naming-objects
+        return self.original_filename.replace("/", "__")
+
+    @property
+    def dt(self) -> pendulum.Date:
+        return self.ts.date()
+
+
+class GTFSScheduleFeedExtractUnzipOutcome(ProcessingOutcome):
+    extract: GTFSFeedExtractInfo = Field(..., exclude={"config"})
+    zipfile_files: Optional[List[str]]
+    zipfile_dirs: Optional[List[str]]
+    extracted_files: Optional[List[GTFSScheduleFeedFile]]
+
+
+class ScheduleUnzipResult(PartitionedGCSArtifact):
```

Comment:
I'm starting to think some of these (successes, failures, etc.) could be put into a Result base type

Reply:
do you want me to do anything differently here?

Reply:
Nope just a general comment